### PR TITLE
Add menu button support for target and lightbox

### PR DESCRIPTION
**Pull request by maccabeelevine**
Add two configurable options to links generated by the menu-button component:

- 'external' for adding target="_blank"
- 'lightbox' for data-lightbox="true"

These support the component being more useful for custom menus.

### CODE REVIEW

File: themes/bootstrap3/templates/_ui/components/menu-button.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/_ui/components/menu-button.phtml:35]: Might it make sense to allow the target attribute to be set directly here instead of abstracting it with a boolean `external`? I admit that I can't really think of a use case for setting a different target, but thought I'd raise the question while we're working on it, just in case there's a reason to do it. Another option might be to allow the `external` option to accept a boolean or a string, and use the string directly but convert boolean true to `_blank`.

**Review by demiankatz — COMMENTED** [review]
This looks good to me, but one thing that might be worth considering before we merge...

File: themes/bootstrap3/templates/_ui/components/menu-button.phtml
Comment (maccabeelevine) [reply themes/bootstrap3/templates/_ui/components/menu-button.phtml:35]: The truth is I had the same train of thought but I also couldn't see any other use case for the target attribute.  I think if we really wanted to generalize, I would do so not by generalizing that particular attribute (i.e. `$current['target']`) but with single `$current['linkAttributes']` array where we just output whatever is contained as attributes.  But then that makes it slightly heavier to actually use the thing, as opposed to the (I think) much more common use case of external=true for target="_blank".  I don't really have a strong opinion though.

Comment (demiankatz) [reply themes/bootstrap3/templates/_ui/components/menu-button.phtml:35]: Fair enough. I suppose for now we can merge this as-is, and if a use case arises, we can always introduce linkAttributes and come up with a way to reconcile that with other settings. At that point, it might make sense to refactor the code to use the `htmlAttributes()` helper, since array manipulation would make reconciliation simpler.

**Review by demiankatz — APPROVED** [review]

File: themes/bootstrap3/templates/_ui/components/menu-button.phtml
Comment (EreMaijala) [reply themes/bootstrap3/templates/_ui/components/menu-button.phtml:35]: I think that _not_ allowing an arbitrary target is a good thing. It's way too easy to mistype `_blank` and get a named window that causes all sorts of trouble.

**Pull request merged**
